TFNetwork.py

```diff
@@ -1964,14 +1964,23 @@ def __init__(self, filename, saveable_params, params_prefix="", load_if_prefix="
     self.params_prefix = params_prefix
     self.load_if_prefix = load_if_prefix
     self.saveable_params = []
+    self.param_mapping = {}
```

Comment:
This also makes it more complicated.

Reply:
Also, it's not documented. What is the type of this?

Reply:
It's a dictionary `new layer name => old layer name`. I'll add the documentation. 
I'm not sure how we can avoid the additional logic. We could omit this attribute, but then we have to rely on the user defined `layer_mapping`. This has two downsides:
 - the user would have to specify the mapping for all parameters of a layer, not just once
 - layers that are not to be renamed are not part of `layer_mapping`. This is the case e.g. if we just want to perform preloading without renaming. 
Using `self.param_mapping`, we can fix this, by checking whether the parameter if part of any renamed layer, and mapping it to itself otherwise.

Reply:
If we have to make a tradeoff between easier code and making sth easier for the user, but for some very rarely used feature, I would very much vote for the easier code.